### PR TITLE
[Modules] API Management allowed values update

### PR DESCRIPTION
**Pull request by clarked-msft**
# Description

- Update apim api type allowed values to include graphql and websocket.
 
## Pipeline references

[ApiManagement - Service](https://github.com/clarked-msft/ResourceModules/actions/workflows/ms.apimanagement.service.yml)

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/ApiManagement/service/main.bicep [hunk 1]
```diff
@@ -111,16 +111,18 @@ param zones array = []
 
 @description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource.')
 @allowed([
-  'allLogs'
-  'GatewayLogs'
+  []
```
Comment (eriqua) [inline modules/ApiManagement/service/main.bicep:114]: Diagnostic settings are implemented consistently across all modules supporting them.

@clarked-msft could you please open an issue in the board to propose the reasons behind the change for the team to discuss and if accepted to be implemented across the library?
I'd suggest to remove the diagnostic settings change from the scope of this PR and keep the `apiType` allowed values update.

Comment (clarked-msft) [reply modules/ApiManagement/service/main.bicep]: I've removed the diagnostic settings change and will open an issue.

Comment (eriqua) [reply modules/ApiManagement/service/main.bicep]: Thanks @clarked-msft. Looks good. 
Please solve merge conflicts and I'll go ahead approving the PR.

File: modules/api-management/service/apis/main.bicep [hunk 2]
```diff
@@ -15,8 +15,10 @@ param apiRevisionDescription string = ''
 
 @description('Optional. Type of API to create. * http creates a SOAP to REST API * soap creates a SOAP pass-through API.')
```
Comment (eriqua) [inline modules/api-management/service/apis/main.bicep:16]: Description would need an update to
```suggestion
@description('Optional. Type of API to create. * http creates a REST API * soap creates a SOAP pass-through API * websocket creates websocket API * graphql creates GraphQL API.')
```

File: modules/ApiManagement/service/main.bicep (hunk 1 above)
Comment (eriqua) [reply modules/ApiManagement/service/main.bicep]: @clarked-msft description should be updated too as per suggestion and readme regenerated after that. Sorry if I missed it in the first review.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**